frontend/src/components/PostList.tsx

```diff
@@ -0,0 +1,179 @@
+            <div>
+              <p className="font-bold">{post.user}</p>
+              <CardTitle className="text-sm text-muted-foreground">
+                Posted on {new Date(post.createdAt).toLocaleString()}
+                {post.createdAt !== post.updatedAt && (
+                  <span>
+                    {" "}
+                    (Edited on {new Date(post.updatedAt).toLocaleString()})
+                  </span>
+                )}
+              </CardTitle>
+            </div>
+            {post.userId === currentUserId && (
+              <DropdownMenu
+                open={openMenus[post.id]}
+                onOpenChange={(isOpen) => handleOpenChange(post.id, isOpen)}
+              >
+                <DropdownMenuTrigger asChild>
+                  <Button variant="ghost" size="sm">
+                    <MoreHorizontal className="h-4 w-4" />
+                    <span className="sr-only">Open menu</span>
+                  </Button>
+                </DropdownMenuTrigger>
+                <DropdownMenuContent align="end" className="w-[200px]">
+                  <DropdownMenuLabel>Actions</DropdownMenuLabel>
+                  <DropdownMenuGroup>
+                    <DropdownMenuItem onClick={() => handleEdit(post)}>
+                      Edit
+                    </DropdownMenuItem>
+                    <DropdownMenuSeparator />
+                    <DropdownMenuItem
+                      className="text-red-600"
+                      onClick={() => confirmDelete(post.id)}
+                    >
+                      Delete
+                    </DropdownMenuItem>
+                  </DropdownMenuGroup>
+                </DropdownMenuContent>
+              </DropdownMenu>
+            )}
+          </CardHeader>
+          <CardContent>
+            <p>{post.content}</p>
+          </CardContent>
+        </Card>
+      ))}
+      {selectedPost && (
+        <Dialog open={openEdit} onOpenChange={setOpenEdit}>
+          <PostContentModal
+            content={content}
+            open={openEdit}
+            onOpenChange={setOpenEdit}
+            setContent={setContent}
+            isSubmitting={isSubmitting}
+            handleSubmit={handleSubmit}
+            title="Edit post"
+          />
+        </Dialog>
+      )}
+      <AlertDialog open={deleteDialogOpen} onOpenChange={setDeleteDialogOpen}>
+        <AlertDialogContent>
+          <AlertDialogHeader>
+            <AlertDialogTitle>Confirm Delete</AlertDialogTitle>
+            <AlertDialogDescription>
+              Are you sure you want to delete this post? This action cannot be
+              undone
+            </AlertDialogDescription>
+          </AlertDialogHeader>
+          <AlertDialogFooter>
+            <AlertDialogCancel>Cancel</AlertDialogCancel>
+            <AlertDialogAction onClick={handleDelete}>Delete</AlertDialogAction>
+          </AlertDialogFooter>
+        </AlertDialogContent>
+      </AlertDialog>
```

Comment:
Could be separate and reusable - for that we should pass not only onOpenChange and open to params, but also onCancel and onDelete